### ISSUE TITLE
Enhancement: [consistent-generic-constructors] check default parameter

### ISSUE THREAD

**Issue by omril1** [typescript-eslint/typescript-eslint#6151]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-generic-constructors/

### Description

I think the rule should check function and constructor arguments with default parameter values as well

Related rule: [no-inferrable-types](https://typescript-eslint.io/rules/no-inferrable-types)

### Fail

```typescript
/* eslint "@typescript-eslint/consistent-generic-constructors": ["error", "constructor"] */
class Foo1 {
  bar(baz: Set<string> = new Set()) { }
}

class Foo2 {
  constructor(bar: Set<string> = new Set()) { }
}

/* eslint "@typescript-eslint/consistent-generic-constructors": ["error", "type-annotation"] */
class Foo1 {
  bar(baz = new Set<string>()) { }
}

class Foo2 {
  constructor(bar = new Set<string>()) { }
}
```

### Pass

```typescript
/* eslint "@typescript-eslint/consistent-generic-constructors": ["error", "constructor"] */
class Foo1 {
  bar(baz = new Set<string>()) { }
}

class Foo2 {
  constructor(bar = new Set<string>()) { }
}

/* eslint "@typescript-eslint/consistent-generic-constructors": ["error", "type-annotation"] */
class Foo1 {
  bar(baz: Set<string> = new Set()) { }
}

class Foo2 {
  constructor(bar: Set<string> = new Set()) { }
}
```

**Comment by Josh-Cena** [comment]
Same as https://github.com/typescript-eslint/typescript-eslint/issues/5285, this will simply be an enhancement without a new option. How does that sound?
